### ISSUE TITLE
Regression: Comparing bytes and string

### ISSUE THREAD

**Issue by Nudin** [mediawiki-utilities/python-mwoauth#35]
When using the code from the [tutorial on wikitech](https://wikitech.wikimedia.org/wiki/Help:Toolforge/My_first_Flask_OAuth_tool) the authentication worked with v0.3.2 but doesn't work anymore with 0.3.5 and fails with:
`Unexpect request token key b"xxxx", expected "xxxx"`

`request_token_key` is a bytestring by definition, while request_token.key is a string.

**Comment by consideRatio** [comment]
In 0.3.5 I've also ended up with issues that possible are caused for the same reason. There were no issues in 0.3.3.

## Initial error

```python
    def identify(mw_uri, consumer_token, access_token, leeway=10.0,
                 user_agent=defaults.USER_AGENT):
        """
        Gather identifying information about a user via an authorized token.
    
        :Parameters:
            mw_uri : `str`
                The base URI of the MediaWiki installation.  Note that the URI
                should end in ``"index.php"``.
            consumer_token : :class:`~mwoauth.ConsumerToken`
                A token representing you, the consumer.
            access_token : :class:`~mwoauth.AccessToken`
                A token representing an authorized user.  Obtained from
                `complete()`
            leeway : `int` | `float`
                The number of seconds of leeway to account for when examining a
                tokens "issued at" timestamp.
    
        :Returns:
            A dictionary containing identity information.
        """
    
        # Construct an OAuth auth
        auth = OAuth1(consumer_token.key,
                      client_secret=consumer_token.secret,
                      resource_owner_key=access_token.key,
                      resource_owner_secret=access_token.secret)
    
        # Request the identity using auth
        r = requests.post(url=mw_uri,
                          params={'title': "Special:OAuth/identify"},
                          auth=auth,
                          headers={'User-Agent': user_agent})
    
        # Special:OAuth/identify unhelpfully returns 200 status even when there is
        # an error in the API call. Check for error messages manually.
        if r.content.startswith(b'{'):
            try:
                resp = r.json()
                if 'error' in resp:
                    raise OAuthException(
                        "A MediaWiki API error occurred: {0}"
                        .format(resp['message']))
            except ValueError as e:
                raise OAuthException(
                    "An error occurred while trying to read json " +
                    "content: {0}".format(e))
        else:
            raise OAuthException(
                "Could not read response from 'Special:OAuth/identify'.  " +
                "Maybe your MediaWiki is not configured correctly?  " +
>               "Expected JSON but instead got: {0!r}".format(r.content[:100]))
E           mwoauth.errors.OAuthException: Could not read response from 'Special:OAuth/identify'.  Maybe your MediaWiki is not configured correctly?  Expected JSON but instead got: b'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Indhc2giLCJhdWQiOiJjbGllbnRfaWQiLCJpc3MiOiJodHR'

/home/erik/anaconda3/lib/python3.7/site-packages/mwoauth/functions.py:263: OAuthException
```

## It is base64 encoded, unexpectedly

```shell
echo "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Indhc2giLCJhdWQiOiJjbGllbnRfaWQiLCJpc3MiOiJodHR" | base64 -d
{"typ":"JWT","alg":"HS256"}
```

I think the breaking change comes from 7f1da20

**Comment by halfak** [comment]
Thanks for the report.  However it looks like these bugs are different.

**Comment by halfak** [comment]
I just released version 0.3.6.  Please let me know if this fixes the issue for you.
